### PR TITLE
Web worker support for auth-next

### PR DESCRIPTION
**Pull request by avolkovi**

### CODE REVIEW

File: packages-exp/auth-exp/demo/rollup.config.js [hunk 1]
```diff
@@ -40,7 +42,27 @@ const es5Builds = [
...
+  {
+    input: 'src/worker/index.ts',
+    output: [{ file: pkg.worker, format: 'esm', sourcemap: true }],
+    plugins: [...commonPlugins,
+      typescriptPlugin({
+      typescript,
+      tsconfigOverride: {
+        compilerOptions: {
+          lib: [
+            "dom",
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/rollup.config.js:57]: we can remove this once we figure out the navigator issue

File: packages-exp/auth-exp/demo/src/index.js [hunk 2]
```diff
@@ -53,12 +53,12 @@ import {
...
   updatePassword,
   updateProfile,
-  verifyPasswordResetCode,
-  OAuthProvider,
-  signInWithPopup,
-  linkWithPopup,
-  reauthenticateWithPopup,
-  browserPopupRedirectResolver
+  verifyPasswordResetCode
+  // OAuthProvider,
```
Comment (avolkovi) [inline packages-exp/auth-exp/demo/src/index.js:57]: TODO: rebase and uncomment

File: packages-exp/auth-exp/demo/src/worker/index.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  persistence: inMemoryPersistence
+});
+
+/**
+ * Returns a promise that resolves with an ID token if available.
+ * @return {!Promise<?string>} The promise that resolves with an ID token if
+ *     available. Otherwise, the promise resolves with null.
+ */
+function getIdToken(): Promise<string | null> {
+  return new Promise((resolve, _reject) => {
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/demo/src/worker/index.ts:48]: just remove the `_reject`

File: packages-exp/auth-exp/demo/src/worker/index.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+/**
+ * Handles the incoming message from the main script.
+ * @param {!MessageEvent} e The message event received.
+ */
+self.onmessage = (e: MessageEvent) => {
+  // https://github.com/microsoft/TypeScript/issues/12657
+  const ctx = self as DedicatedWorkerGlobalScope;
+  if (e.data && e.data.type) {
```
Comment (samhorlbeck) [inline packages-exp/auth-exp/demo/src/worker/index.ts:163]: `if (e.data?.type)`

File: packages-exp/auth-exp/demo/src/worker/index.ts (hunk 3 above)
Comment (avolkovi) [reply packages-exp/auth-exp/demo/src/worker/index.ts]: done

File: packages-exp/auth-exp/demo/src/worker/index.ts (hunk 4 above)
Comment (avolkovi) [reply packages-exp/auth-exp/demo/src/worker/index.ts]: done

File: packages-exp/auth-exp/demo/src/index.js (hunk 2 above)
Comment (avolkovi) [reply packages-exp/auth-exp/demo/src/index.js]: done

**Pull request merged**